packages/select/src/SelectItem.tsx

```diff
@@ -0,0 +1,44 @@
+import type { SharedSelectState } from './types';
+import type { ItemProps } from '@react-types/shared';
+import type { ComponentPropsWithoutRef, ElementType } from 'react';
+
+type SelectItemProps<T extends object, P extends ElementType> = ItemProps<T> & {
+  as?: P;
+  onClick?: (_e: MouseEvent, state: SharedSelectState) => void;
+  [key: string]: any;
+};
+
+const SelectItem = <T extends object, P extends ElementType = 'li'>(
```

Comment:
style thing: why did you go with an assignment expression instead of a function declaration?

Reply:
I'm not super picky on one vs the other but mainly because I like consistency where possible -- e.g. everything can be an assignment expression, but not everything can be a function declaration. Which do you prefer?

Reply:
PS you made me notice a bug by commenting on this line 😂 `ElementType = 'li'` should be `ElementType = 'div'`

Reply:
:success:

I tend to prefer `function` when defining something a function because it maps nicely to a human description of the code, but that's not always "convenient" so I'm also not super picky. In this case, I was curious because the type parameter TypeScript syntax for arrow functions always takes me a second to understand. 😅